### DISCUSSION TITLE
Perform brute-force vector search while keeping index

### DISCUSSION THREAD

**Discussion by ramialt** [milvus-io/milvus#24804] [Q&A and General discussion] [answered]
Hello,
As I understand it, only one index type is allowed on a field at a time.
My usual use-case is to use HNSW index to perform vector search on my data. However, I would sometimes like to perform a brute-force KNN search for experimentation purposes. Is this possible without having to destroy and later re-create my HNSW index? If so, how?

P.S. From reading the documents, it seems that "brute-force search" == "exhaustive search" == "Search with FLAT 'index'". Is this correct?

**Comment by yhmo** [comment]
To brute-force search, declare a FLAT index:
```
index_params = {"index_type": "FLAT", "metric_type": "L2", "params": {}}
collection.create_index(field_name="embeddings", index_params=index_params)
```

For IVF index, we can set "nprobe" equal to "nlist" so that it search all buckets just like a brute-force search.

For HNSW index, you can try setting "ef" to a huge value(I am not sure it can work):
```
    results = collection.search(data=search_vectors,
                                anns_field="embeddings",
                                param={"metric_type": "L2", "params": {"ef": 1000000}},
                                limit=5)
```

**Reply by ramialt** [reply]
The upper bound of "ef" [seems to be 32768](https://milvus.io/docs/index.md#HNSW), and efConstruction even less at 512.

So then if I already have created an HNSW index, there is no way to brute-force search my data without deleting the HNSW index? Other than making a gigantic copy of all of my data to a new collection.

**Reply by yhmo** [reply, marked as answer]
> The upper bound of "ef" [seems to be 32768](https://milvus.io/docs/index.md#HNSW), and efConstruction even less at 512.
> 
> So then if I already have created an HNSW index, there is no way to brute-force search my data without deleting the HNSW index? Other than making a gigantic copy of all of my data to a new collection.

I think so.

**Reply by ramialt** [reply]
Created feature request for this at https://github.com/milvus-io/milvus/issues/24843
